java/client/src/main/java/compatibility/clients/jedis/Jedis.java

```diff
@@ -387,4 +408,1394 @@ public Set<String> keys(String pattern) {
+                        } catch (NumberFormatException e) {
+                            // Skip invalid limit parameters
+                        }
+                    }
+                    break;
+
+                case "GET":
+                    if (i + 1 < params.length) {
+                        builder.getPattern(params[++i]);
+                    }
+                    break;
+
+                case "ASC":
+                    builder.orderBy(SortBaseOptions.OrderBy.ASC);
+                    break;
+
+                case "DESC":
+                    builder.orderBy(SortBaseOptions.OrderBy.DESC);
+                    break;
+
+                case "ALPHA":
+                    builder.alpha();
+                    break;
+
+                default:
+                    // Unknown parameter, skip it
+                    break;
+            }
+        }
+
+        return builder.build();
+    }
+
+    /**
+     * Serialize a key's value.
+     *
+     * @param key the key
+     * @return the serialized value, or null if key does not exist
+     */
+    public byte[] dump(String key) {
+        checkNotClosed();
+        try {
+            return glideClient.dump(GlideString.of(key)).get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new JedisException("DUMP operation failed", e);
+        }
+    }
+
+    /**
+     * Deserialize a value and store it at a key.
+     *
+     * @param key the key
+     * @param ttl time to live in milliseconds (0 for no expiration)
+     * @param serializedValue the serialized value
+     * @return "OK"
+     */
+    public String restore(String key, long ttl, byte[] serializedValue) {
+        checkNotClosed();
+        try {
+            return glideClient.restore(GlideString.of(key), ttl, serializedValue).get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new JedisException("RESTORE operation failed", e);
+        }
+    }
+
+    /**
+     * Move a key to another Redis instance.
+     *
+     * @param host destination host
+     * @param port destination port
+     * @param key the key to migrate
+     * @param destinationDb destination database
+     * @param timeout timeout in milliseconds
+     * @return "OK" or "NOKEY"
+     */
+    public String migrate(String host, int port, String key, int destinationDb, int timeout) {
+        checkNotClosed();
+        try {
+            Object result =
+                    glideClient
+                            .customCommand(
+                                    new String[] {
+                                        "MIGRATE",
+                                        host,
+                                        String.valueOf(port),
+                                        key,
+                                        String.valueOf(destinationDb),
+                                        String.valueOf(timeout)
```

Comment:
Do we need to validate these inputs (such as negative values for port)? Does Glide validate this? Does Jedis?